src/modules/taskList.js

```diff
@@ -1,19 +1,99 @@
-const toDoList = [
-  {
-    description: 'Organize digital files',
-    completed: false,
-    index: 1,
-  },
-  {
-    description: 'Clean the house',
-    completed: false,
-    index: 2,
-  },
-  {
-    description: 'Plan a day trip',
-    completed: false,
-    index: 3,
-  },
-];
-
-export default toDoList;
+import saveTasks from './localStorage.js';
+
+export default function renderTaskList(tasks) {
+  const deleteTask = (task) => {
+    const taskIndex = tasks.findIndex((t) => t.index === task.index);
+    if (taskIndex !== -1) {
+      tasks.splice(taskIndex, 1);
+
+      for (let i = taskIndex; i < tasks.length; i += 1) {
+        tasks[i].index -= 1;
+      }
+
+      saveTasks(tasks);
+      renderTaskList(tasks);
+    }
+  };
+
+  const makeTaskDescriptionEditable = (taskDescription, task) => {
+    taskDescription.contentEditable = true;
+    taskDescription.focus();
+    taskDescription.classList.add('editable');
+
+    const originalDescription = task.description;
+
+    taskDescription.addEventListener('keydown', (event) => {
+      if (event.key === 'Enter') {
+        event.preventDefault();
+        taskDescription.blur();
+      } else if (event.key === 'Escape') {
+        event.preventDefault();
+        taskDescription.innerText = originalDescription;
+        taskDescription.blur();
+      }
+    });
+
+    taskDescription.addEventListener('blur', () => {
+      const newDescription = taskDescription.innerText.trim();
+      taskDescription.contentEditable = false;
+
+      if (newDescription !== '') {
+        task.description = newDescription;
+        saveTasks(tasks);
+        renderTaskList(tasks);
+      } else {
+        taskDescription.innerText = originalDescription;
+      }
+    });
+  };
+
+  const taskList = document.getElementById('task-list');
+  taskList.innerHTML = ''; // Clear previous list
+
+  tasks.forEach((task) => {
+    const listItem = document.createElement('li');
+
+    const checkbox = document.createElement('input');
+    checkbox.type = 'checkbox';
+    checkbox.checked = task.completed;
+    listItem.appendChild(checkbox);
+
+    const taskDescription = document.createElement('span');
+    taskDescription.innerText = task.description;
+    if (task.completed) {
+      taskDescription.classList.add('completed');
+    }
+    listItem.appendChild(taskDescription);
+
+    const kebabMenuIcon = document.createElement('img');
+    kebabMenuIcon.src = './../assets/kebab-menu.png';
+    kebabMenuIcon.alt = 'Kebab Menu';
+    kebabMenuIcon.classList.add('kebab-menu');
+    listItem.appendChild(kebabMenuIcon);
+
+    const trashCanIcon = document.createElement('img');
+    trashCanIcon.src = './../assets/trash-can-icon.png';
+    trashCanIcon.alt = 'Trash Can';
+    trashCanIcon.classList.add('trash-can-icon');
+    trashCanIcon.style.display = 'none';
+    listItem.appendChild(trashCanIcon);
```

Comment:
[OPTIONAL] Please fix your image patch because your images are not displayed so kindly fix it :+1: check below :point_down: your images which are not displayed :slightly_smiling_face: 

![Screenshot from 2023-06-18 02-32-52](https://github.com/tasnimchy96/to-do-list.github.io/assets/45226848/d2eac724-ef5d-4ed8-97ce-c6e9b67531e7)